### ISSUE TITLE
Galois conjugate objects: Hilbert, Bianchi, EC

### ISSUE THREAD

**Issue by davidfarmer** [LMFDB/lmfdb#2262]
If you start browsing Hilbert modular forms (just look at the first two on Qsqrt5)
you will find that often they occur in pairs that have the same L-function.
Similarly for the associated elliptic curves.  (This is one of those things that
seems surprising/wrong the first time you see it, but then it seems normal
after you think for a while.)

The reason is that those objects are Galois conjugates of each other,
and the L-function is defined over Q.  Another way to say it is that
both Hilbert modular forms on GL(2)/F lift to the same thing on GL(4)/Q.
And the Weil restrictions of the two elliptic curves are the same.

Since the two HMFs (or the two ECNFs) are genuinely different,
they could be considered to be friends of each other.  And all of those
things should be friends of the L-function.

So, this is a request to have Galois conjugate things be friends of each other.

**Comment by JohnCremona** [comment]
I certainly agree that this is desirable and was thinking about how to implement it.  Surely best to do a one-off computation which results in each of these objects storing a list of the labels of its Galois conjugates.  The one-off code will take each object (e.g. an elliptic curve) defined over the base field K, compute its conjugates and look for that object in the database, adding the label of that to the list for the original object.  So one sub-problem to be solved here -- which we have not yet had to deal with for any of these types -- is, given the usual data we have on an object (Weierstrass equation, or level ideal + some Hecke eigenvalues), find it in the database.  That is a function which it would be good to have anyway.

The second issue is, what to do about base fields which are not Galois.  If K is a non-Galois cubic then it has no automorphisms, but there are two other conjugate number fields inside the sextic Galois closure.  So would we then say that each object over K has no Galois conjugates?  I think we would have to; the alternative would be that the conjugates would be the same as the original abstractly but with a different embedding of K into C.  Now if you go to the home page of such a field (e.g. http://beta.lmfdb.org/NumberField/3.1.23.1) it says that the field is not Galois, but says nothing else about its real/complex embeddings.  There's a case for adding the infinite prime to the list of "Frobenius cycle types" which already, for finite p, give links to the associated localizations.  I don't think that R or C have home pages (yet)!

**Comment by davidfarmer** [comment]
> The second issue is, what to do about base fields which are not Galois. If K is a non-Galois cubic then it
> has no automorphisms, but there are two other conjugate number fields inside the sextic Galois closure. So
> would we then say that each object over K has no Galois conjugates?

The motivation for this issue, which seems to be insensitive to any
details about the underlying number field, is that two "different"
objects can have the same L-function.  By 'same' I mean really
literally exactly the same.

This can happen because L-functions are defined over Q.

If you have an elliptic curve over a quadratic field, then its
L-function may arise as a degree 2 Euler product over the primes
of that field, but it multiplies out to be a degree 4 Euler
product over Q.  (which usually is primitive.)

Not matter what the field is, you can group the terms in the
Euler product according to the rational prime they lie over,
and get an Euler product over Q, right?  And the terms multiply
out to give a polynomial in p^{-s} with p\in \Q, right?  Am I
missing something?

Maybe "Galois conjugate" isn't the right name, but the fact that
two curves have the same L-function is in interesting relationship.

**Comment by JohnCremona** [comment]
You are quite right.  For any elliptic curve E over any number field K of degree d, the product of the Euler factors at all K-primes above the Q-prime p will be a polynomial in p^{-s} of degree 2d (or less for bad primes).
Secondly, applying an automorphism of K may (and usually will) give a different curve E but its L-function will be the same (as an L-function over Q).

e.g. 1. For K quadratic there really are pairs of curves with the same L-function (apart from curves which are base-change from Q where both are the same), and we agree that each curve in the pair should known about its twin (as a friend to be displayed).
e.g.2. If K is a Galois cubic then there will similarly be triples of curves all with the same L-function.
e.g.3. But if K is a non-Galois cubic then there will be no "friends" of this kind.

So the conclusion is that the size of these sets of mutual friends will be equal to the number of automorphisms of K (or in degenerate cases of base-change, a factor of that number), rather than the degree of K.  And we should identify them and show them!

**Comment by edgarcosta** [comment]
Shouldn't we just do that through the database instead of coding it?

**Comment by jvoight** [comment]
FWIW, @AndrewVSutherland worked out the following theorem (that will appear in our genus 2 automorphy paper, god willing):

Theorem: For i = 1, 2, let A_i be an abelian variety over a number field F_i.  Let B_i = Res_{F_i/QQ} A_i be the restriction of scalars of A_i from F_i to QQ. Then the following are equivalent:
(i) L(A1, s) = L(A2, s);
(ii) L(B1, s) = L(B2, s); and
(iii) B1 is isogenous to B2 over QQ.

So David, your (analytic) L-function criterion is equivalent to a (geometric) isogeny class criterion, but for the restriction of scalars to QQ.  

I think more friends is better--note to self, I need to get out more--so this doesn't bear on the technical question about how to display and deal with this in the LMFDB--but at least you shouldn't expect anything beyond what you already have observed.

**Comment by JohnCremona** [comment]
@edgarcosta "Shouldn't we just do that through the database instead of coding it?" -- yes, this info should be stored in the database.  To get it there some work, i.e. coding is needed.

**Comment by edgarcosta** [comment]
@JohnCremona Sorry for not being clear. Yes, I totally agree with you. I was trying to say that I would prefer us to avoid coding each particular case on the website code. Instead, I would prefer  us to have, for this kind of friendship, something as simple as: "if they have the same L-function, then they are friends"

**Comment by AndrewVSutherland** [comment]
This issue is related to #3117 and would ideally be addressed at the same time.  We have already taken some steps in this direction by having objects look in the lfunc_instances table to find other objects with the same L-function when creating their lists of friends (as of Release 1.1, this is done for CMFs, BMFs, HMFs, ECs, and genus 2 curves, although not all the BMF,HMF,ECNF L-functions have been computed).
